core/store/src/trie/iterator.rs

```diff
@@ -91,6 +94,16 @@ impl<'a> TrieIterator<'a> {
             TrieIterator::Memtrie(iter) => iter.seek_prefix(key),
         }
     }
+
+    /// Skips the next node in the iterator without recording its value.
+    /// As a safety check, the caller must know the key of the item to
+    /// be skipped. (e.g., for continuing a batched iteration).
+    pub fn skip_one(&mut self, expected_key: &[u8]) {
+        match self {
+            TrieIterator::Disk(iter) => iter.skip_one(expected_key),
+            TrieIterator::Memtrie(iter) => iter.skip_one(expected_key),
```

Comment:
I don't understand why we have two separate arms, which call the same implementation. I see the same above for seek_prefix. Maybe it's some language limitation I'm not aware of.

Reply:
Since TrieIterator is an enum, to get access to `DiskTrieIterator` or `STMemTrieIterator` it's deconstructed each time. I'm not sure if there is a better pattern we could use?

Reply:
Yeah, but it only matters that they implement `TrieIteratorImpl`, it doesn't matter for `TrieIterator` what is the exact implementation. But I suspect there is no way to store a field `ineer: &dyn TrieIteratorImpl` because `TrieIteratorImpl` is generic.